pylint/checkers/base.py

```diff
@@ -1772,9 +1777,28 @@ def _check_misplaced_constant(self, node, left, right, operator):
         self.add_message('misplaced-comparison-constant', node=node,
                          args=(suggestion,))
 
+    def _check_logical_tautology(self, node):
+        # import pdb;pdb.set_trace()
+        left_operand = node.left
+        right_operand = node.ops[0][1]
+        operator = node.ops[0][0]
+        if (isinstance(left_operand, astroid.Const)
```

Comment:
I will check, but AFAIR, literal-comparision didn't check it.

Reply:
$ cat sample.py
if 786 == 786:
    pass
elif 1 == 2 and 666 != 666:
    pass
elif 666 > 666:
    pass
elif 666 <= 666:
    pass
elif 786 is 786:
    pass
elif 786 is not 786:
    pass

$ pylint sample.py
Using config file /home/sussolan/git_repos/pylint/pylintrc
************* Module pylint.sample
R:  1, 3: Logical tautology in comparison - 786 == 786 (logical-tautology)
R:  3,16: Logical tautology in comparison - 666 != 666 (logical-tautology)
R:  5, 5: Logical tautology in comparison - 666 > 666 (logical-tautology)
R:  7, 5: Logical tautology in comparison - 666 <= 666 (logical-tautology)
R:  9, 5: Logical tautology in comparison - 786 is 786 (logical-tautology)
R:  9, 5: Comparison to literal (literal-comparison)
R: 11, 5: Logical tautology in comparison - 786 is not 786 (logical-tautology)
R: 11, 5: Comparison to literal (literal-comparison)

Few points to note here:
1) literal-comparison only checks for identity operators.
   How to handle cases like "786 == 786", should it be handled in this checker or literal-comparison or skipped altogether?
2) Also regarding your other comment to remove argument from message, error for line 3 shows explicitly which comparison has issue, so if any statement has many comparisons, it will help in identifying the exact problematic comparison. 
However if you still think otherwise, I will remove it from message.

Reply:
1) Ah sorry I had forgotten the purpose of literal-comparison. I think what you have is fine.

2) Good point. I'm happy for the argument to stay.